### ISSUE TITLE
allow "scroll spy" with the current section highlighted

### ISSUE THREAD

**Issue by samuelcolvin** [squidfunk/mkdocs-material#1309]
__I checked that...__

- [x] ... the documentation does not mention anything about my idea
- [x] ... to my best knowledge, my idea wouldn't break something for other users
- [x] ... there are no open or closed issues that are related to my idea

## Description

It would be great if an extra attribute was set on `md-nav__link` to allow custom CSS to change how the page TOC looks to highlight the current item, rather than blur items above the current page y-offset.

This could be implemented by something like adding `data-md-state="current"` to the TOC item corresponding to the top section visible on the screen.

### Use Cases

I'd like to highlight the page-TOC item corresponding to the section currently on screen, similar to:
* bootstrap 3's scrollspy: https://getbootstrap.com/docs/3.3/javascript/#scrollspy (see the right hand TOC on that page, as well as that section of the docs)
* simpler example: https://juliabyexample.helpmanual.io/

I'm not suggesting editing the standard CSS and thereby enforcing a change for most users, just allow users to implement this themselves by adding css like:

```css 
.md-nav__link[data-md-state=blur] {
    color: rgba(0,0,0,.87);  /* remove blurring */
}
.md-nav__link[data-md-state=current] {
    font-weight: 700;
}
```

**Comment by samuelcolvin** [comment]
Should involve a relatively small change to 

https://github.com/squidfunk/mkdocs-material/blob/5ad1c7031dc595ab4bd0eb01615639c8224d701c/src/assets/javascripts/components/Material/Nav/Blur.js#L90-L113

Should be entirely backwards compatible.

**Comment by squidfunk** [comment]
Exact duplicate of #1102 ;-) Will be fixed with #1306, but this will take some time.

**Comment by samuelcolvin** [comment]
ha, looked but couldn't find a similar issue.
